src/test/ui/issue-15034.nll.stderr

```diff
@@ -5,12 +5,12 @@ LL |         Parser { lexer: lexer }
    |         ^^^^^^
 
 error[E0621]: explicit lifetime required in the type of `lexer`
-  --> $DIR/issue-15034.rs:27:25
+  --> $DIR/issue-15034.rs:27:9
    |
 LL |     pub fn new(lexer: &'a mut Lexer) -> Parser<'a> {
    |                ----- consider changing the type of `lexer` to `&'a mut Lexer<'a>`
 LL |         Parser { lexer: lexer }
-   |                         ^^^^^ lifetime `'a` required
+   |         ^^^^^^^^^^^^^^^^^^^^^^^ lifetime `'a` required
```

Comment:
These seem less good but still "tolerable" — I think with a bit more thought we should be able to narrow back down to the `lexer` expression. The problem is that we still have trouble distinguishing "intermediate, passthrough" regions (like the one on `Parser`) from the "source regions" (the ones on `lexer`). That's ok for now though I think.